packages/flame/lib/src/components/mixins/notifiable.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:meta/meta.dart';
+
+mixin Notifiable on Component {
+  FlameGame get _gameRef {
+    final game = findGame();
+    assert(
+      game == null || game is FlameGame,
+      "Notifiable can't be used without FlameGame",
+    );
+    return game! as FlameGame;
+  }
+
+  @override
+  @mustCallSuper
+  void onMount() {
+    _gameRef.notifiers[runtimeType]?.add(this);
+  }
+
+  @override
+  @mustCallSuper
+  void onRemove() {
+    _gameRef.notifiers[runtimeType]?.remove(this);
```

Comment:
This should use a method on the gameRef so we dont need to use runtimeType:

```dart
class FlameGame {
	void addNotifier(Component component) {
	  	for (final notifier in notifiers) {
			if (notifier.valid(component) {
  	     			notifier.add(component);
  	   		}
  		}
 	}
  
  ...
}

class ComponentNotifier<T extends Component> {
 	bool valid(Component c) => c is T;
  
  ...
}
```

This (pseudo) code should allow for sub classes as well and we dont need to depend on runtimeType

Reply:
This changes the semantics though: if `A` extends `B`, then `A() is B`, but `A().runtimeType != B`. But maybe it's for the better?

Reply:
I wonder what is the benefits of this change, in a first look, I believe that iterating through a list and checking if they are valid is more expensive than just a map access and also more complex code than the current one.

Can you elaborate more on what benefits this change will bring before I go with it?